pkg/image/prune/prune.go

```diff
@@ -770,6 +744,54 @@ func pruneStreams(
 	return nil
 }
 
+// pruneISTagHistory processes tag event list of the given image stream tag. It removes references to images
+// that are going to be removed or are missing in the graph.
+func pruneISTagHistory(
+	g graph.Graph,
+	prunableImageNodes map[string]*imagegraph.ImageNode,
+	keepYoungerThan time.Time,
+	streamName string,
+	imageStream *imageapi.ImageStream,
+	tag string,
+) (tagUpdated, tagDeleted bool) {
+	history := imageStream.Status.Tags[tag]
+	newHistory := imageapi.TagEventList{}
+
+	for i, tagEvent := range history.Items {
+		glog.V(4).Infof("Checking tag event %d with image %q", i, tagEvent.Image)
+
+		if _, ok := prunableImageNodes[tagEvent.Image]; ok {
+			glog.V(4).Infof("Image stream tag %s:%s revision %d - removing because image %q matches deleted image", streamName, tag, i, tagEvent.Image)
+			tagUpdated = true
+			continue
+		}
+		n := imagegraph.FindImage(g, tagEvent.Image)
+		if n != nil {
+			glog.V(4).Infof("Image stream tag %s:%s revision %d - keeping because image %q is not deleted", streamName, tag, i, tagEvent.Image)
+			newHistory.Items = append(newHistory.Items, tagEvent)
+			continue
+		}
+
+		if n == nil && !tagEvent.Created.After(keepYoungerThan) {
+			glog.V(4).Infof("Image stream tag %s:%s revision %d - deleting because image %q is absent", streamName, tag, i, tagEvent.Image)
+			tagUpdated = true
+			continue
+		}
+		glog.V(4).Infof("Image stream tag %s:%s revision %d - keeping because of age", streamName, tag, i)
```

Comment:
But it isn't kept.

Can you extract this part into `func tagEventIsPrunable(tagEvent image.TagEvent, g, prunableImageNodes, keepYoungerThan) (ok bool, reason string)`?

Reply:
> But it isn't kept.

Good catch, extracted.